src/main/java/org/folio/event/service/UserTenantService.java

```diff
@@ -94,12 +96,15 @@ public Future<Boolean> deleteUserTenant(UserTenant userTenant, String tenantId,
   }
 
   public Future<Void> validateUserAcrossTenants(User entity, User userFromStorage, Map<String, String> okapiHeaders, Conn conn, Context vertxContext) {
-    Predicate<User> predicate = user -> ObjectUtils.notEqual(user.getUsername(), userFromStorage.getUsername());
+    Predicate<User> predicate = user -> ((UserType.STAFF.getTypeName().equals(user.getType()) && StringUtils.isNotBlank(user.getUsername())) ||
+      ObjectUtils.notEqual(UserType.STAFF.getTypeName(), user.getType())) &&
+      ObjectUtils.notEqual(user.getUsername(), userFromStorage.getUsername());
```

Comment:
looks over-complicated, I think better to add this check in another method where you checking that username is blank
```suggestion
   if (StringUtils.isBlank(entity.getUsername()) && UserType.STAFF.getTypeName().equals(entity.getType())) {
      logger.error(USERNAME_IS_NOT_POPULATED);
      return Future.failedFuture(USERNAME_IS_NOT_POPULATED);
    }
    if (predicate.test(entity)) {
      return isUsernameUniqueAcrossTenants(entity.getUsername(), consortiaCentralTenantId, okapiHeaders, vertxContext);
    }
```